### ISSUE TITLE
Allow embeding libpython to fail

### ISSUE THREAD

**Issue by ktbarrett** [cocotb/cocotb#2272]
Currently in  the embed library the routine that loads and initializes the Python interpreter (`void embed_init_python(void)`) is not allowed to fail. Obviously it can fail, hence the current obtuse work around. Once embed is merged into the GPI, it should be allowed to fail and have the workaround removed.

**Comment by ktbarrett** [comment]
This has been obviated by the fact the embed library has been removed.
